### PR TITLE
statistics: extract the common function to rebuild the queue

### PR DESCRIPTION
**Pull request by Rustin170506**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/55906

Problem Summary:

### What changed and how does it work?

This pull request extracted a function for rebuilding the job queue, allowing us to reuse code to implement the new priority queue.

Check out https://github.com/pingcap/tidb/pull/55889/files#r1774853942 to see how we use it.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Rustin170506 — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

File: pkg/statistics/handle/autoanalyze/priorityqueue/queue.go [hunk 1]
```diff
@@ -31,8 +186,9 @@ func NewAnalysisPriorityQueue() *AnalysisPriorityQueue {
 }
 
 // Push adds a job to the priority queue with the given weight.
-func (apq *AnalysisPriorityQueue) Push(job AnalysisJob) {
+func (apq *AnalysisPriorityQueue) Push(job AnalysisJob) error {
```
[the remark this answers is not in the document]
Comment (Rustin170506) [reply pkg/statistics/handle/autoanalyze/priorityqueue/queue.go:189]: Another implementation will result in an error. So we need to keep them using the same singature.

**Review by AilinKid — APPROVED** [review]

**Review by time-and-fate — APPROVED** [review]

**Pull request merged**